Cpp2IL.Decompiler/IL/MemoryAddress.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+
+namespace Cpp2IL.Decompiler.IL;
+
+/// <summary>
+/// Memory operand in the format of [base+addend+index*scale].
+/// </summary>
+public struct MemoryAddress(object? baseRegister = null, object? indexRegister = null, long addend = 0, int scale = 0)
```

Comment:
Are `baseRegister` and `indexRegister` always `Register` or `null`? If so, they should be typed as `Register?` to avoid boxing.